protocol/nats_jetstream/v3/options_test.go

```diff
@@ -0,0 +1,341 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
```

Comment:
nit: many of these tests don't assert error cases. If you run your tests with coverage, you'll see that several code paths are not checked

Reply:
added more unit tests and added extra asserts